src/bp-moderation/classes/suspend/class-bp-suspend-member.php

```diff
@@ -406,6 +427,28 @@ public function manage_unhidden_member( $member_id, $hide_sitewide, $force_all,
 			);
 			$bp_background_updater->save()->schedule_event();
 		}
+
+		// Update friend count.
+		if ( bp_is_active( 'friends' ) ) {
+			$friend_ids = friends_get_friend_user_ids( $member_id );
+
+			if ( ! empty( $friend_ids ) ) {
+				$friend_ids = array_filter( array_unique( $friend_ids ) );
+				if ( $this->background_disabled || ! $force_bg_process ) {
+					$this->bb_update_member_friend_count( $member_id, $friend_ids, 'unhidden' );
+				} else {
+					$bp_background_updater->data(
+						array(
+							array(
+								'callback' => array( $this, 'bb_update_member_friend_count' ),
+								'args'     => array( $member_id, $friend_ids, 'unhidden' ),
+							),
+						)
+					);
+					$bp_background_updater->save()->dispatch();
+				}
+			}
+		}
```

Comment:
please remove this schedule and move the code to `get_related_contents` based on the action params.